### PR TITLE
test: add comprehensive integration tests for browser agent

### LINKED ISSUE #1047: test: add integration tests for browser agent

### PR DESCRIPTION
**Pull request by Achieve3318**
Fixes #1047

This PR adds comprehensive integration tests for the browser agent, covering initialization, toolkit integration, browser operations, search functionality, note-taking, error handling, and end-to-end workflows.

### New Test File
- `backend/tests/integration/agents/test_browser_agent.py` (1,248 lines)
  - **TestBrowserAgentInitialization**: Tests agent setup and configuration
  - **TestBrowserAgentToolkitIntegration**: Tests integration with all toolkits
  - **TestBrowserAgentBrowserOperations**: Tests browser operation capabilities
  - **TestBrowserAgentSearchIntegration**: Tests search functionality
  - **TestBrowserAgentNoteTakingIntegration**: Tests note-taking features
  - **TestBrowserAgentEndToEndWorkflows**: Tests complete workflows
  - **TestBrowserAgentErrorHandling**: Tests error handling and edge cases
  - **TestBrowserAgentSystemMessage**: Tests system message configuration
  - **TestBrowserAgentToolNames**: Tests tool names configuration
  - **TestBrowserAgentRealWorldScenarios**: Tests real-world scenarios

## Test Coverage

The integration tests cover:
- ✅ Agent initialization with various configurations
- ✅ Integration with HybridBrowserToolkit, SearchToolkit, NoteTakingToolkit, TerminalToolkit, and HumanToolkit
- ✅ Browser operations: visit_page, click, type, scroll, navigation, console, sheet operations, page snapshots
- ✅ Search workflows with and without Google search
- ✅ Note-taking with URL citations
- ✅ End-to-end research workflows
- ✅ Error handling for invalid URLs, network errors, missing elements, CAPTCHA challenges, and timeouts
- ✅ System message and tool configuration
- ✅ Real-world scenarios (marked as very_slow)

All tests are properly marked with `@pytest.mark.integration` and use appropriate fixtures from conftest.py.

## Testing

Run the integration tests with:
```bash
pytest backend/tests/integration/agents/test_browser_agent.py -m integration
```

For very slow tests:
```bash
pytest backend/tests/integration/agents/test_browser_agent.py --full-test-mode
```

### CODE REVIEW

**Comment by Achieve3318** [comment]
Hi, @Wendong-Fan I'd love to get your review and approval
Thank you for your time

**Comment by Achieve3318** [comment]
Hi, @fengju0213 
could you review my PR?
thank you.

File: backend/tests/integration/agents/test_browser_agent.py [hunk 1]
```diff
@@ -0,0 +1,1248 @@
...
+from typing import Dict, Any, List
+from unittest.mock import AsyncMock, MagicMock, patch, Mock
+import pytest
+
+from camel.agents import ChatAgent
+from camel.messages import BaseMessage
+from camel.models import BaseModelBackend
+from camel.responses import ChatAgentResponse
+
+from app.utils.agent import browser_agent, ListenChatAgent
```
Comment (LuoPengcheng12138) [inline backend/tests/integration/agents/test_browser_agent.py]: Failed to run: `browser_agent`, `ListenChatAgent` and `Agents` are no longer present in the imported modules, which may be due to updates on the `main` branch.

Comment (Achieve3318) [reply backend/tests/integration/agents/test_browser_agent.py]: I have fixed that, Can you check again, please?

File: backend/app/utils/agent.py
Comment (LuoPengcheng12138) [inline backend/app/utils/agent.py]: hi, What is the relationship between this file and `backend/app/agent/agent_model.py`? It appears to be a duplicate file？

**Comment by Achieve3318** [comment]
Hi, @LuoPengcheng12138 
could you check again please?
Thank you for your review

File: backend/app/utils/agent.py
Comment (Wendong-Fan) [inline backend/app/utils/agent.py:1]: why we need this file?

File: backend/tests/integration/agents/test_browser_agent.py [hunk 2]
```diff
@@ -0,0 +1,1250 @@
...
+            project_id="test_project_456",
+            email="test@example.com",
+            question="Test question",
+            model_type="gpt-4",
+            model_platform="openai",
+            api_key="test_key",
+            api_url="https://api.openai.com/v1",
+        )
+
+        with patch('app.utils.agent.env', return_value='9222'), \
```
Comment (Wendong-Fan) [inline backend/tests/integration/agents/test_browser_agent.py:67]: should we patch from `app.agent.factory.browser` ?

File: backend/tests/integration/agents/test_browser_agent.py [hunk 3]
```diff
@@ -0,0 +1,1250 @@
...
+            call_args = mock_agent_model.call_args
+            assert call_args is not None
+
+
+@pytest.mark.integration
+class TestBrowserAgentBrowserOperations:
+    """Test browser agent browser operation capabilities."""
+
+    @pytest.mark.asyncio
+    async def test_browser_agent_visit_page_operation(self, temp_dir, mock_model_backend):
```
Comment (Wendong-Fan) [inline backend/tests/integration/agents/test_browser_agent.py:421]: how would this test agent really visited page operation only with `assert agent is not None`?

**Comment by Achieve3318** [comment]
Hi, @Wendong-Fan 
If I fix the issue, can you reopen this again?
